### PR TITLE
prov/rxd, verbs-ud: error handling, size and prefix size fixes

### PR DESCRIPTION
**Pull request by aingerson**
rxd:
- error handling
- set tx/rx size correctly
- separate tx/rx prefix sizes

verbs:
- remove tx prefix size requirement

### CODE REVIEW

File: prov/rxd/src/rxd_cq.c [hunk 1]
```diff
@@ -1101,6 +1114,22 @@ void rxd_handle_recv_comp(struct rxd_ep *ep, struct fi_cq_msg_entry *comp)
...
+
+	ret = fi_cq_readerr(ep->dg_cq, &err, 0);
+	if (ret < 0) {
+		FI_WARN(&rxd_prov, FI_LOG_CQ,
+			"Error reading CQ: %s\n", fi_strerror(-ret));
+	} else {
+		FI_WARN(&rxd_prov, FI_LOG_CQ,
+			"Received %s error from core provider: %s\n",
+			err.flags & FI_SEND ? "tx" : "rx", fi_strerror(-err.err)); 
+	}
```
Comment (shefty) [inline prov/rxd/src/rxd_cq.c]: You should probably fail the transfer for send failures.  But this shouldn't happen either.

File: prov/rxd/src/rxd_ep.c [hunk 2]
```diff
@@ -1252,8 +1252,9 @@ int rxd_endpoint(struct fid_domain *domain, struct fi_info *info,
 	rxd_ep->prefix_size = dg_info->ep_attr->msg_prefix_size;
 	fi_freeinfo(dg_info);
 
-	rxd_ep->rx_size = info->rx_attr->size;
-	rxd_ep->tx_size = info->tx_attr->size;
+	rxd_ep->rx_size = MIN(dg_info->rx_attr->size, info->rx_attr->size);
+	rxd_ep->tx_size = MIN(dg_info->tx_attr->size, info->tx_attr->size);
```
Comment (shefty) [inline prov/rxd/src/rxd_ep.c]: Why is this needed?  The info sizes should already be adjusted based on the dgram size.  The app expects to use the info sizes itself.

File: prov/rxd/src/rxd_cq.c (hunk 1 above)
Comment (aingerson) [reply prov/rxd/src/rxd_cq.c]: Yeah this is mainly there as a precaution for warning of any failures (which I added in trying to find that verbs memory corruption bug wondering if maybe the transfer completed in an error but I just didn't catch it). I can't see it ever getting here but figured to add it just in case.
As for failing the transfer - I considered that as well but it gets a bit trickier for larger sends with a lot of packets (ie, sequence numbers, do we clear the future packets because the message already failed, do we have to wait for completions per our in_use check discussion). All the extra implementation/handling of this didn't seem worth it to me for a situation that shouldn't happen. But I can add all that in if you prefer.

File: prov/rxd/src/rxd_ep.c (hunk 2 above)
Comment (shefty) [reply prov/rxd/src/rxd_ep.c]: With the proposed changes, the code looks wrong.  The app is expecting to send/receive the info size values.  Those should already be set based on the underlying dg_info sizes. I don't understand what the failure actually is
